quickwit-cli/src/lib.rs

```diff
@@ -293,6 +301,37 @@ pub async fn delete_index_cli(args: DeleteIndexArgs) -> anyhow::Result<()> {
     Ok(())
 }
 
+pub async fn garbage_collect_index_cli(args: GarbageCollectIndexArgs) -> anyhow::Result<()> {
+    debug!(
+        index_uri = %args.index_uri,
+        "garbage-collect-index"
+    );
+    quickwit_telemetry::send_telemetry_event(TelemetryEvent::GarbageCollect).await;
+
+    let (metastore_uri, index_id) =
+        extract_metastore_uri_and_index_id_from_index_uri(&args.index_uri)?;
+    let affected_files = garbage_collect_index(metastore_uri, index_id).await?;
+
+    let deleted_bytes: u64 = affected_files
+        .iter()
+        .map(|entry| entry.file_size_in_bytes)
+        .sum();
+
+    if deleted_bytes > 0 {
```

Comment:
Can we do the test on affect_files being empty or not? (I'm not sure if files with 0 byte are allowed or not on S3 but it seems more "logical")